IoTHubBuddy/IoTViewer/HubPage.xaml.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using IoTHubBuddy.Models;
+
+// The Blank Page item template is documented at http://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace IoTHubBuddy
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class HubPage : Page
+    {
+        private IoTAccountData data;
+        private ICollection<EventHubData> hubs;
+        public HubPage()
+        {
+            this.InitializeComponent();
+            
+        }
+        protected override async void OnNavigatedTo(NavigationEventArgs e)
+        {
+            IoTAccountData account = e.Parameter as IoTAccountData;
+            if(checkValidity(account))
+            {
+                
+                data = account;
+                hubs = await IoTDataManager.GetIoTHubs(data.Subscription);
+                if(hubs.Count == 0)
+                {
+                    showError("You have no IoTHubs registered under this subscription.");
+                } else
+                {
+                    hideErrors();
+                    foreach (EventHubData h in hubs)
+                    {
+                        HubList.Items.Add(h.HubName);
+                    }
```

Comment:
unfortunately, ItemCollection does not have an addRange method or the equivalent. I cannot think of another way to iterate over the whole list without using a for loop